src/main/java/io/mosip/iiitb/odte/consumers/OnDemandTemplateExtractionConsumerImpl.java

```diff
@@ -0,0 +1,198 @@
+package io.mosip.iiitb.odte.consumers;
+
+import com.fasterxml.jackson.core.JsonProcessingException;
+import com.google.inject.Inject;
+import io.mosip.iiitb.odte.IssueCredentialsConf;
+import io.mosip.iiitb.odte.config.OnDemandAppConfig;
+import io.mosip.iiitb.odte.dto.CredentialRequestAdditionalDataDto;
+import io.mosip.iiitb.odte.dto.IssueCredentialsRawResponseDto;
+import io.mosip.iiitb.odte.dto.IssueCredentialsResponseDto;
+import io.mosip.iiitb.odte.lib.ApiRequestService;
+import io.mosip.iiitb.odte.lib.MosipResponseError;
+import io.mosip.iiitb.odte.utils.SaltUtil;
+import org.slf4j.Logger;
+
+import java.io.IOException;
+import java.security.NoSuchAlgorithmException;
+
+import static io.mosip.iiitb.odte.utils.Utilities.generateIdHash;
+
+public class OnDemandTemplateExtractionConsumerImpl  implements EventConsumer<DecryptedOnDemandTemplateRecord> {
+
+    private final ApiRequestService apiRequestService;
+    private final SaltUtil saltUtil;
+    private final OnDemandAppConfig config;
+
+    private final Logger logger;
+
+    @Inject
+    public OnDemandTemplateExtractionConsumerImpl(
+            ApiRequestService apiRequestService,
+            SaltUtil saltUtil,
+            OnDemandAppConfig config,
+            Logger logger
+    ) {
+        this.logger = logger;
+        this.apiRequestService = apiRequestService;
+        this.saltUtil = saltUtil;
+        this.config = config;
+    }
+
+    @Override
+    public EventConsumerResponse processRecord(DecryptedOnDemandTemplateRecord record) {
+        String uid = record.getId();
+        IssueCredentialsConf issueCredentialsAuthConf = new IssueCredentialsConf();
+        issueCredentialsAuthConf.setAppId(config.issueCredsAppId());
+        issueCredentialsAuthConf.setClientId(config.issueCredsClientId());
+        issueCredentialsAuthConf.setClientPass(config.issueCredsClientPass());
+        issueCredentialsAuthConf.setIssuer(config.issueCredsPartnerCode());
+
+        this.logger.debug(
+            "issue CredentialsAuthConf = {}",
+            issueCredentialsAuthConf
+        );
+        String appId = config.regprocAppId();
+        String clientId = config.regprocClientId();
+        String clientPass = config.regprocClientPass();
+        String partnerCode = config.regprocPartnerCode();
+        String authToken = null;
+        try {
+            authToken = apiRequestService.getAuthToken(
+                appId,
+                clientId,
+                clientPass
+            );
+            logger.debug(
+                String.format(
+                    "authToken = %s\n",
+                    authToken
+                )
+            );
```

Comment:
why you want to get a new auth token for every message, can we not get it once and use it till the expiry ?

Reply:
Well, it does get it from cache if its cached already, it doesn't get a new auth token every time. That logic is in a ApiRequestService.java, consumer shouldn't be bothering how it gets the auth token, it should just get the auth token